.github/workflows/pr_build.yaml

```diff
@@ -0,0 +1,102 @@
+name: Build Binaries (Pull Request)
+
+on:
+  pull_request:
+
+env:
+  BUILD_TYPE: RelWithDebInfo
+
+permissions:
+  contents: write
+
+jobs:
+  gen-build-version:
+    name: Generate Build Version
+    runs-on: ubuntu-latest
+    outputs:
+      build: ${{ steps.build_name.outputs.build }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Get the cargo package version
+        run: |
+          VERSION=$(awk -F '"' '/^version =/ { print $2 }' Cargo.toml)
+          echo "The package version is $VERSION"
+          echo "CARGO_PKG_VERSION=$VERSION" >> $GITHUB_ENV
```

Comment:
I think the variable we export here should just be called something like "NEXT_VERSION". It's technically gonna be the next version of LTR we release. it's only an implementation detail right now that we extract this from the Cargo.toml